pkg/api/serialization_test.go

```diff
@@ -174,7 +174,8 @@ func TestCommonKindsRegistered(t *testing.T) {
 				t.Error(err)
 			}
 			defaults := gv.WithKind("")
-			if _, got, err := api.Codecs.LegacyCodec().Decode([]byte(`{"kind":"`+kind+`"}`), &defaults, nil); err != nil || gvk != *got {
+			var got *schema.GroupVersionKind
+			if obj, got, err = api.Codecs.LegacyCodec().Decode([]byte(`{"kind":"`+kind+`"}`), &defaults, obj); err != nil || gvk != *got {
```

Comment:
That's the point: the doc of the versioning codec says that internal objects are returned. The code also implement that.

The interface docs (https://github.com/kubernetes/kubernetes/blob/6a4d5cd7cc58e28c20ca133dab7b0e9e56192fe3/staging/src/k8s.io/apimachinery/pkg/runtime/interfaces.go#L55) sound like the `defaults` gvk is effective.

What is right?